clients/src/main/java/org/apache/kafka/clients/consumer/internals/Acknowledgements.java

```diff
@@ -207,6 +233,7 @@ public List<ShareAcknowledgeRequestData.AcknowledgementBatch> getShareAcknowledg
         if (acknowledgements.isEmpty()) return batches;
         ShareAcknowledgeRequestData.AcknowledgementBatch currentBatch = null;
         Iterator<Map.Entry<Long, AcknowledgeType>> iterator = acknowledgements.entrySet().iterator();
+        int maxRecordsWithSameAcknowledgeType = 0;
```

Comment:
I think it would be better if the variable `maxRecordsWithSameAcknowledgeType` wasn't almost identical to the constant `MAX_RECORDS_WITH_SAME_ACKNOWLEDGE_TYPE`.